src/pudl/settings.py

```diff
@@ -71,64 +72,103 @@ def validate_tables(cls, tables):  # noqa: N805
         return sorted(set(tables))
 
 
-def create_dataset_settings(name: str) -> GenericDatasetSettings:
+class Ferc1Settings(GenericDatasetSettings):
     """
-    Create a pydantic settings class for a dataset.
+    An immutable pydantic model to validate Ferc1Settings.
 
-    This function dynamically creates subclasses of GenericDatasetSettings
-    using information from the constants WORKING_PARTITIONS and PUDL_TABLES.
-    The returned class can be subclassed to add additional fields or validators.
+    Parameters:
+        years: List of years to validate.
+        tables: List of tables to validate.
 
-    Args:
-        name: the name of the dataset.
+        working_partitions: Dictionary of working paritions.
+        working_tables: List of working tables.
+    """
 
-    Returns:
-        dataset_model: Subclass of GenericDatasetSettings
+    working_partitions: ClassVar = {
+        "years": tuple(range(1994, 2021))
+    }
+    working_tables: ClassVar = sorted(list(pc.PUDL_TABLES["ferc1"]))
+
+    years: List[int] = working_partitions["years"]
+    tables: List[str] = working_tables
+
+
+class EpaCemsSettings(GenericDatasetSettings):
     """
-    try:
-        partitions = pc.WORKING_PARTITIONS[name]
-        tables = pc.PUDL_TABLES[name]
-    except KeyError:
-        raise KeyError(f"{name} is not an available dataset.")
+    An immutable pydantic model to validate EPA CEMS settings.
 
-    # sort the partitions.
-    tables = sorted(list(tables))
-    partitions = {k: sorted(v) for (k, v) in partitions.items()}
+    Parameters:
+        years: List of years to validate.
+        states: List of states to validate.
+        tables: List of tables to validate.
 
-    class_name = name.title() + "Settings"
-    dataset_model = create_model(class_name,
-                                 working_partitions=(ClassVar, partitions),
-                                 working_tables=(ClassVar, tables),
-                                 tables=(List, tables),
-                                 **partitions,
-                                 __base__=GenericDatasetSettings)
-    return dataset_model
+        working_partitions: Dictionary of working paritions.
+        working_tables: List of working tables.
+    """
 
+    working_partitions: ClassVar = {
+        "years": tuple(range(1994, 2021)),
+        "states": tuple(sorted(EPACEMS_STATES))
```

Comment:
Is there a reason besides immutability to keep the `tuple()` here?

Reply:
Nope, just immutability.